src/audio/google_rtc_audio_processing.c

```diff
@@ -0,0 +1,383 @@
+	cd->output_buffer_index = 0;
+	
+	comp_set_drvdata(dev, cd);
+	dev->state = COMP_STATE_READY;
+	comp_dbg(dev, "google_rtc_audio_processing_create(): Ready");
+	return dev;
+fail:
+	comp_err(dev, "google_rtc_audio_processing_create(): Failed");
+	if (cd) {
+		rfree(cd->output_buffer);
+		rfree(cd->aec_reference_buffer);
+		rfree(cd->raw_mic_buffer);
+		comp_data_blob_handler_free(cd->tuning_handler);
+		rfree(cd);
+	}
+	rfree(dev);
+	return NULL;
+}
+
+static void google_rtc_audio_processing_free(struct comp_dev *dev)
+{
+	struct google_rtc_audio_processing_comp_data *cd = comp_get_drvdata(dev);
+
+	comp_dbg(dev, "google_rtc_audio_processing_free()");
+
+	GoogleRtcAudioProcessingFree(cd->state);
+	cd->state = NULL;
+	rfree(cd->output_buffer);
+	rfree(cd->aec_reference_buffer);
+	rfree(cd->raw_mic_buffer);
+	comp_data_blob_handler_free(cd->tuning_handler);
+	rfree(cd);
+	rfree(dev);
+}
+
+
+static int google_rtc_audio_processing_trigger(struct comp_dev *dev, int cmd)
+{
+	comp_dbg(dev, "google_rtc_audio_processing_trigger(): %d", cmd);
+
+	return comp_set_state(dev, cmd);
+}
+
+static int google_rtc_audio_processing_prepare(struct comp_dev *dev)
+{
+	struct google_rtc_audio_processing_comp_data *cd = comp_get_drvdata(dev);
+	struct comp_buffer *source_buffer;
+	struct list_item *source_buffer_list_item;
+
+	comp_dbg(dev, "google_rtc_audio_processing_prepare()");
+
+	/* searching for stream and feedback source buffers */
+	list_for_item(source_buffer_list_item, &dev->bsource_list) {
+		source_buffer = container_of(source_buffer_list_item, struct comp_buffer,
+						 sink_list);
+		source_buffer = buffer_acquire_irq(source_buffer);
+		if (source_buffer->source->ipc_config.type == SOF_COMP_DEMUX)
+			cd->aec_reference = source_buffer;
+		else if (source_buffer->source->ipc_config.type == SOF_COMP_DAI)
+			cd->raw_microphone = source_buffer;
+		source_buffer = buffer_release_irq(source_buffer);
+	}
+
+	cd->output = list_first_item(&dev->bsink_list, struct comp_buffer, source_list);
+
+	switch (cd->output->stream.frame_fmt) {
+#if CONFIG_FORMAT_S16LE
+	case SOF_IPC_FRAME_S16_LE:
+		break;
+#endif /* CONFIG_FORMAT_S16LE */
+	default:
+		comp_err(dev, "unsupported data format: %d", cd->output->stream.frame_fmt);
+		return -EINVAL;
+	}
+
+	if (cd->output->stream.rate != GOOGLE_RTC_AUDIO_PROCESSING_SAMPLERATE) {
+		comp_err(dev, "unsupported samplerate: %d", cd->output->stream.rate);
+		return -EINVAL;
+	}
+
+	return comp_set_state(dev, COMP_TRIGGER_PREPARE);
+}
+
+static int google_rtc_audio_processing_reset(struct comp_dev *dev)
+{
+	comp_dbg(dev, "google_rtc_audio_processing_reset()");
+
+	return comp_set_state(dev, COMP_TRIGGER_RESET);
+}
+
+static int google_rtc_audio_processing_copy(struct comp_dev *dev)
+{
+	struct google_rtc_audio_processing_comp_data *cd = comp_get_drvdata(dev);
+
+	struct comp_copy_limits cl;
+	int frame;
+	uint32_t aec_reference_buff_frag = 0;
+	uint32_t raw_microphone_buff_frag = 0;
+	uint32_t output_buff_frag = 0;
```

Comment:
please, either remove these initialisations or the ones below at lines 309, 326 and 327